### PR TITLE
Add a pipeline list to the transcoder details dialog

### PR DESCRIPTION
**Pull request by jbroadus**
List current running transcode pipelines. This will be allow the addition of cancel and debug options. Specifically, I plan to add an option to dump the graph of a pipeline in an upcoming change.

### CODE REVIEW

File: src/engines/gstpipelinebase.h [hunk 1]
```diff
@@ -47,4 +48,18 @@ class GstPipelineBase : public QObject {
   const int id_;
 };
 
+class GstPipelineModel : public QStandardItemModel {
+  Q_OBJECT
+
+ public:
+  GstPipelineModel(QObject* parent = nullptr);
```
Comment (hatstand) [inline src/engines/gstpipelinebase.h:55]: `explicit`

File: src/engines/gstpipelinebase.h [hunk 2]
```diff
@@ -47,4 +48,18 @@ class GstPipelineBase : public QObject {
...
+  void AddPipeline(int id, const QString& name);
+  void RemovePipeline(int id);
+
+ private:
+  int FindRowById(int id);
```
Comment (hatstand) [inline src/engines/gstpipelinebase.h:60]: I think this method can be `const`?

File: src/transcoder/transcodedialog.cpp [hunk 3]
```diff
@@ -118,7 +119,7 @@ TranscodeDialog::TranscodeDialog(QWidget* parent)
 }
 
 TranscodeDialog::~TranscodeDialog() {
-  delete log_ui_;
+  delete details_ui_;
```
Comment (hatstand) [inline src/transcoder/transcodedialog.cpp:122]: Can you change these to use `unique_ptr` while you're here?

File: src/transcoder/transcoder.cpp [hunk 4]
```diff
@@ -195,10 +195,16 @@ void Transcoder::JobState::PostFinished(bool success) {
                               new Transcoder::JobFinishedEvent(this, success));
 }
 
+QString Transcoder::JobState::GetDisplayName() {
+  return QFileInfo(job_.input).fileName() + " to " +
```
Comment (hatstand) [inline src/transcoder/transcoder.cpp:199]: If you use an arrow here, (`->` perhaps) then we don't need to worry about translations.

File: src/engines/gstpipelinebase.h (hunk 1 above)
Comment (jbroadus) [reply src/engines/gstpipelinebase.h]: Changed.

File: src/transcoder/transcodedialog.cpp (hunk 3 above)
Comment (jbroadus) [reply src/transcoder/transcodedialog.cpp]: Done.

File: src/transcoder/transcoder.cpp (hunk 4 above)
Comment (jbroadus) [reply src/transcoder/transcoder.cpp]: Changed to use `=>`.

File: src/engines/gstpipelinebase.h (hunk 2 above)
Comment (jbroadus) [reply src/engines/gstpipelinebase.h]: Yep. Fixed.

**Pull request merged**